src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -14,211 +24,3020 @@ namespace Microsoft.CodeAnalysis.CSharp
         }
 
-        protected virtual void VisitCatchBlock(BoundCatchBlock catchBlock, ref TLocalState finallyState)
+        protected void VisitLvalue(BoundExpression node)
         {
-            if (catchBlock.ExceptionSourceOpt != null)
+            if (_trackRegions && node == this.firstInRegion && this.regionPlace == RegionPlace.Before) EnterRegion();
+            switch (node?.Kind)
             {
-                VisitLvalue(catchBlock.ExceptionSourceOpt);
+                case BoundKind.Parameter:
+                    VisitLvalueParameter((BoundParameter)node);
+                    break;
+
+                case BoundKind.Local:
+                    VisitLvalue((BoundLocal)node);
+                    break;
+
+                case BoundKind.ThisReference:
+                case BoundKind.BaseReference:
+                    // no need for it to be previously assigned: it is on the left.
+                    break;
+
+                case BoundKind.PropertyAccess:
+                    var access = (BoundPropertyAccess)node;
+
+                    if (Binder.AccessingAutoPropertyFromConstructor(access, _member))
+                    {
+                        var backingField = (access.PropertySymbol as SourcePropertySymbol)?.BackingField;
+                        if (backingField != null)
+                        {
+                            VisitFieldAccessInternal(access.ReceiverOpt, backingField);
+                            break;
+                        }
+                    }
+
+                    goto default;
+
+                case BoundKind.FieldAccess:
+                    {
+                        BoundFieldAccess node1 = (BoundFieldAccess)node;
+                        VisitFieldAccessInternal(node1.ReceiverOpt, node1.FieldSymbol);
+                        break;
+                    }
+
+                case BoundKind.EventAccess:
+                    {
+                        BoundEventAccess node1 = (BoundEventAccess)node;
+                        VisitFieldAccessInternal(node1.ReceiverOpt, node1.EventSymbol.AssociatedField);
+                        break;
+                    }
+
+                case BoundKind.TupleLiteral:
+                    ((BoundTupleExpression)node).VisitAllElements((x, self) => self.VisitLvalue(x), this);
+                    break;
+
+                case BoundKind.SuppressNullableWarningExpression:
+                    VisitLvalue(((BoundSuppressNullableWarningExpression)node).Expression);
+                    break;
+
+                default:
+                    VisitRvalue(node);
+                    break;
             }
 
-            if (catchBlock.ExceptionFilterOpt != null)
+            if (_trackRegions && node == this.lastInRegion && this.regionPlace == RegionPlace.Inside) LeaveRegion();
+        }
+
+        protected virtual void VisitLvalue(BoundLocal node)
+        {
+            // no need for it to be previously assigned: it is on the left.
```

Comment:
i don't really understand this comment.  We're in abstract-flow-pass.  but this seems to be referring to assignment stuff...